### PR TITLE
🎉 New Source: TVMaze Schedule

### PR DESCRIPTION
**Pull request by isaacharrisholt**
## What

It can be useful to have TV schedule data to hand when running television marketing campaigns. This PR adds a source - TVMaze - that makes it easy and free to do this for many countries and global stations (e.g. Netflix)

https://github.com/airbytehq/connector-contest/issues/220
## How

Adds a TVMaze Schedule connector with the following streams:

- `domestic` for local, country-specific TV
- `web` for global TV channels
- `future` for all known upcoming programmes (though the other streams can also fetch data for future dates)

## Recommended reading order

1. `tvmaze_schedule.yaml`
2. `spec.yaml`

## 🚨 User Impact 🚨

No changes to existing code.

## Pre-merge Checklist

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

None.

</details>

<details><summary><strong>Integration</strong></summary>

image

</details>

<details><summary><strong>Acceptance</strong></summary>

image

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Thanks for the contribution! Someone will review it during this week. If you have any questions please create comments so the reviewer can focus on those points.

File: airbyte-integrations/connectors/source-tvmaze-schedule/source_tvmaze_schedule/tvmaze_schedule.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    request_options_provider:
+      request_parameters:
+        country: |
+          {{
+            config['domestic_schedule_country_code']
+            if options['name'] == 'domestic' 
+            else config['web_schedule_country_code'].replace('global', '')
+            if options['name'] == 'web' 
+            else ''
+          }}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-tvmaze-schedule/source_tvmaze_schedule/tvmaze_schedule.yaml:19]: not sure if this the best frmat, why not overwrite the value directly in the stream to remove all this `if` clauses?

File: airbyte-integrations/connectors/source-tvmaze-schedule/source_tvmaze_schedule/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      description: Country code for domestic TV schedule retrieval.
+      examples:
+        - US
+        - GB
+    web_schedule_country_code:
+      type: string
+      description: |
+        Country code for web TV schedule retrieval. Leave blank for all
+        countries plus global web channels (e.g. Netflix). Alternatively, set
+        to 'global' for just global web channels.
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-tvmaze-schedule/source_tvmaze_schedule/spec.yaml:33]: The ISO 3166-1 code country. Please add this info.

Can you add the default to `global` and as example?

File: airbyte-integrations/connectors/source-tvmaze-schedule/integration_tests/catalog.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "streams": [
+    {
+      "name": "domestic",
+      "supported_sync_modes": ["full_refresh"],
+      "source_defined_cursor": true,
+      "default_cursor_field": "airdate",
+      "json_schema": {}
+    },
+    {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-tvmaze-schedule/integration_tests/catalog.json:10]: Remove this file.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
A few comments, please add the doc with the changelog.
Can you share the credentials for testing?

File: airbyte-integrations/connectors/source-tvmaze-schedule/source_tvmaze_schedule/spec.yaml (hunk 2 above)
Comment (isaacharrisholt) [reply airbyte-integrations/connectors/source-tvmaze-schedule/source_tvmaze_schedule/spec.yaml:33]: I've added the ISO code, but not convinced by having global as the default, as leaving it blank is also a valid option that provides the most data.

File: airbyte-integrations/connectors/source-tvmaze-schedule/integration_tests/catalog.json (hunk 3 above)
Comment (isaacharrisholt) [reply airbyte-integrations/connectors/source-tvmaze-schedule/integration_tests/catalog.json:10]: Done

File: airbyte-integrations/connectors/source-tvmaze-schedule/source_tvmaze_schedule/tvmaze_schedule.yaml (hunk 1 above)
Comment (isaacharrisholt) [reply airbyte-integrations/connectors/source-tvmaze-schedule/source_tvmaze_schedule/tvmaze_schedule.yaml:19]: I tried overriding this in the stream's `$options`, but it seems that it doesn't accept the `InterpolatedString` type, as it just used the string `{{ config['domestic_schedule_country_code'] }}` as the request parameter

**Comment by isaacharrisholt** [comment]
Replied to comments and docs added. No credentials needed as this uses the free TVMaze API which doesn't need an API key @marcosmarxm

**Comment by marcosmarxm** [comment]
Thanks @isaacharrisholt I'll review this contribution tomorrow!

**Review by marcosmarxm — APPROVED** [review]
congrats @isaacharrisholt I'll publish later today.

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**